src/main/analytics/webdav-handlers.ts

```diff
@@ -1,38 +1,92 @@
-import { FileDownloadedDomainEvent } from '../../workers/webdav/modules/files/domain/FileDownloadedDomainEvent';
-import { FileDeletedDomainEvent } from '../../workers/webdav/modules/files/domain/FileDeletedDomainEvent';
+import { ipcWebdav, IpcWebdavFlow, IpcWebdavFlowErrors } from '../ipcs/webdav';
+import {
+  trackHandledWebdavError,
+  trackWebdavError,
+  trackWebdavEvent,
+} from './service';
 import { WebdavErrorContext } from '../../shared/IPC/events/webdav';
 
-function subscribeToDomainEvents() {
-  function subscribeTo<Event extends WebdavDomainEvent>(
-    eventName: Event['eventName'],
-    listener: (
-      event: IpcMainEvent,
-      domainEvent: ReturnType<Event['toPrimitives']>
-    ) => void
-  ) {
-    ipcMain.on(`webdav.${eventName}`, listener);
-  }
+function subscribeToFlowEvents(ipc: IpcWebdavFlow) {
+  ipc.on('WEBDAV_FILE_DELETED', (_, payload) => {
+    const { name, type, size } = payload;
 
-  subscribeTo<FileCreatedDomainEvent>(
-    FileCreatedDomainEvent.EVENT_NAME,
-    (_, attributes) => {
-      trackWebdavEvent('Upload', attributes);
-    }
-  );
+    trackWebdavEvent('Delete', {
+      name,
+      type,
+      size,
+    });
+  });
+
+  ipc.on('WEBDAV_FILE_DOWNLOADED', (_, payload) => {
+    const { name, type, size, uploadInfo } = payload;
+
+    trackWebdavEvent('Upload', {
+      name,
+      type,
+      size,
+      elapsedTime: uploadInfo.elapsedTime,
+    });
+  });
+
+  ipc.on('WEBDAV_FILE_MOVED', (_, payload) => {
+    const { name, folderName } = payload;
 
-  // For some reason FileDownloadedDomainEvent.EVENT_NAME does not work (￣(エ)￣)ゞ
-  subscribeTo<FileDownloadedDomainEvent>('file.downloaded', (_, attributes) => {
-    trackWebdavEvent('Preview', attributes);
+    trackWebdavEvent('Move', {
+      name,
+      folderName,
+    });
   });
 
-  subscribeTo<FileDeletedDomainEvent>(FileDeletedDomainEvent.EVENT_NAME, () => {
-    trackWebdavEvent('Delete', {});
+  ipc.on('WEBDAV_FILE_OVERWRITED', (_, payload) => {
+    const { name } = payload;
+
+    trackWebdavEvent('Move', {
+      name,
+    });
+  });
+
+  ipc.on('WEBDAV_FILE_RENAMED', (_, payload) => {
+    const { name } = payload;
+
+    trackWebdavEvent('Rename', {
+      name,
+    });
+  });
+
+  ipc.on('WEBDAV_FILE_CLONNED', (_, payload) => {
+    const { name, type, size, uploadInfo } = payload;
+
+    trackWebdavEvent('Upload', {
+      name,
+      type,
+      size,
+      clonned: true,
+      elapsedTime: uploadInfo.elapsedTime,
+    });
+  });
+
+  ipc.on('WEBDAV_FILE_UPLOADED', (_, payload) => {
+    const { name, type, size, uploadInfo } = payload;
+
+    trackWebdavEvent('Upload', {
+      name,
+      type,
+      size,
+      elapsedTime: uploadInfo.elapsedTime,
+    });
+  });
+}
+
+function subscribeToFlowErrors(ipc: IpcWebdavFlowErrors) {
+  ipc.on('WEBDAV_FILE_UPLOADED_ERROR', (_, payload) => {
+    const { name, error } = payload;
+    trackWebdavError('Upload Error', { name, error });
```

Comment:
maybe add `reportError` call too so we have the data available on Sentry?

Reply:
The report to Sentry is made on the Webdav worker, the analytics need to be on the main process as it relies on a NodeJs process